src/workloads/execution/DensifyTimeseriesCollection.yml

```diff
@@ -0,0 +1,451 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of $densify stage with a numeric step
+  in timeseries collections.
+
+GlobalDefaults:
+  dbname: &db test
+  batchSize: &batchSize 30000
+  fieldName: &field "numeric"
+  unitName: &unit null
+  index: &index
+    keys: {numeric: 1}
+    options: {name: "numeric"}
+  smallStepSize: &smallStep {^RandomInt: {min: 3, max: 3}}
+  largeStepSize: &largeStep {^RandomInt: {min: 8, max: 8}}
+  explicitBounds: &bounds [{^RandomInt: {min: 5000, max: 15000}}, {^RandomInt: {min: 20000, max: 40000}}]
+  nop: &Nop {Nop: true}
+
+Actors:
+- Name: CreateTimeSeriesCollection
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: *db
+    Operation:
+      OperationMetricsName: CreateTimeSeriesCollection
+      OperationName: RunCommand
+      OperationCommand:
+        {create: &coll Collection0, timeseries: {timeField: "timestamp", metaField: "numeric"}}
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - Repeat: 1
+  - *Nop
+
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10000
+    BatchSize: *batchSize
+    Document:
+      partitionKey: {^RandomInt: {min: 1, max: 100}}
+      timestamp: {^RandomDate: {}}
```

Comment:
Should this have a range?

Reply:
RandomDate doesn't require a range, and since we aren't worried about repeating values I wasn't worried about leaving it like this.  I think since we are densifying the meta field, this is fine?

Reply:
You raise an interesting point. Tbh I don't know much about the use cases of time series collections, but it feels like we should probably be operating on the timestamp field? Should we have at least one case where we densify on it?

Reply:
done